### ISSUE TITLE
Extension should be able to contribute a task template and a user can pick a template

### ISSUE THREAD

**Issue by egamma** [Microsoft/vscode#966]
Currently the Configure Task Runner action generates a large task.json file with many different task configurations, that the user has to clean-up eventually, that is,  delete the configs that do not apply.

Instead of generating one large file we should allow the user to select what kind of task that should be configured and then to only generate this one. This is similar to how we generate the launch configurations.

**Comment by stkb** [comment]
It would be great if extensions could hook into this too to provide configurations and a custom tasks.json. The same goes with the launch.json too.

**Comment by danielschmitz** [comment]
Let me suggest a simple example to implementation  (tasks.json) ....

```
{
        // ROOT WITHOUT COMMAND
	"version": "0.1.0",
	"isShellCommand": true,
	"showOutput": "silent",
	"args": [],
	
	"tasks": [
		{
             //CHILDREN WITH COMMAND ;)
            "taskName": "Build Type Script",
            "suppressTaskName": true,
            "isBuildCommand": true,
			"command": "tsc",
            "args": ["tsc -p ."]
        },
        {
            "taskName": "Open Browser",
            "suppressTaskName": true,
			"command": "live-server",
            "args": ["--open public_html"]
        }
	]
}
```

AND CTRL+SHIFT+B:

```
{ "key": "ctrl+shift+b",          "command": "workbench.action.tasks.runTask" }
```

**Comment by usagi** [comment]
// Note: Related #981

**Comment by egamma** [comment]
Extracted comment from #2263

Yes, please!  With the PowerShell extension we want to provide a default task to run Pester tests.  It would be great if an extension could provide the initial tasks.json file you get with `Tasks: Configure Task Runner`.  We would also like to provide a problemMatcher that could be referenced from the Pester (test) task.  The problemMatcher is somewhat lengthy and it would be nice to reference it like so:
```javascript
{
	"version": "0.1.0",

	// Start PowerShell
	"command": "${env.windir}\\sysnative\\windowspowershell\\v1.0\\PowerShell.exe",

	// The command is a shell script
	"isShellCommand": true,

	// Show the output window always
	"showOutput": "always",

	// Allow Pester to invoke scripts and run Pester
	"args": [
        "-NoProfile",
        "Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope Process;",
        "Write-Host 'Invoking Pester...'; Invoke-Pester;",
        "Invoke-Command { Write-Host \"Completed all tasks in task runner: $($args[0])\" } -args"
        ],

    // Associate with test task runner
    "tasks": [
        {
            "taskName": "Pester",
            "isTestCommand": true,
            "problemMatcher": "$pester"
         }
    }
}
```
Seems like the `package.json` needs a `contributes.Tasks` where extension authors can contribute both problemWatchers and task initial configurations.

**Comment by dbaeumer** [comment]
Here is what I did for the Februar milestone: the monolithic tasks.json example got removed in favor of using snippets. However due to a limitation we have right now we can't contribute a snippet for specific files from the outside. We will look into this in the next milestone.

![capture](https://cloud.githubusercontent.com/assets/1931590/13149435/cf995810-d661-11e5-9d17-db7060d93aa2.png)

**Comment by dbaeumer** [comment]
See https://github.com/Microsoft/vscode/issues/3125

**Comment by rkeithhill** [comment]
I like the direction but yeah, we need to be able to provide task snippets with our extension and have those show up when the user edits a tasks.json file. How would you handle duplicates?  Say another extension created a BeterTypeScriptTasks.json that contained the snipper `tsc - watch`?  Allow both but show the origin (extension) of each snippet?

**Comment by fbricon** [comment]
This feature would be interesting for the vscode-java extension, giving the ability to contribute [gradle task templates](https://github.com/redhat-developer/vscode-java/issues/103), for instance.

**Comment by dbaeumer** [comment]
[Addressed in new terminal runner]

In the terminal runner world extension can contribute tasks programmatically. So the Java extension can contribute a gradle task dynamically by checking if a gradle configuration is available without having the user to edit the tasks.json file.

**Comment by dbaeumer** [comment]
Closing. Got addressed in task 2.0.0
